x/evm/module.go

```diff
@@ -261,7 +274,13 @@ func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.Val
 				ctx.Logger().Error("failed to send wei surplus of %s to EVM module account", surplusWei)
 			}
 		}
+
+		if evmHooks != nil && evmHooks.OnBalanceChange != nil && (surplusUsei.GT(sdk.ZeroInt()) || surplusWei.GT(sdk.ZeroInt())) {
+			evmModuleAddress := am.keeper.AccountKeeper().GetModuleAddress(types.ModuleName)
+			tracers.TraceBlockReward(ctx, evmHooks, am.keeper.BankKeeper(), evmModuleAddress, am.keeper.GetEVMAddressOrDefault(ctx, evmModuleAddress), surplusUsei, surplusWei)
```

Comment:
the nature of `surplus` here is closer to `burnt` actually, since a surplus is resulted from a debit and a smaller credit. The difference between the two iiuc, in ethereum's world, would be considered as burnt